algolia-crawler.js

```diff
@@ -39,6 +39,16 @@ new Crawler({
             lvl2: 'article h2',
             lvl3: ['article h3', 'article h4'], // an array is a list of fallbacks. Authors sometimes omit h3 to skip the side nav.
             lvl4: ['article h4', 'article h5'],
+            products: {
```

Comment:
@gabriele-ct 
this structure is called custom fields in algolia which then allows you to make Filter queries, which can work. 
another approach can also be the Tags feature which is used by the reserved word `_tags:['product1','product2']`

we will need to compare which of them are better in terms of our requirements and the requirement that might happen in the future (having multiple levels of filtering)
would it better be served by custom Filters or Tags

Reply:
"levels" as in hierarchy are not something the kind of engine algolia is does OOTB,  it can be emulated through multiple fields each of which represents a level.   See how the heading levels are modeled in docsearch.  

But I would not focus on it yet, the hieararchy style filtering Ux has many issues and I am not convinced it is sustainable.   Focus on properly modeling the multi-value product attribute first, that's more important (e.g. yo may have to use tags for that and a field for the breadcrumb and content type if there are no multi-value fields available.

Reply:
@FFawzy 
certainly it makes more sense to use tags. I didn't know the API, I'll dig a bit more into it